ruby-builder.js

```diff
@@ -9,23 +9,29 @@ const rubyBuilderVersions = require('./ruby-builder-versions')
 const builderReleaseTag = 'enable-shared'
 const releasesURL = 'https://github.com/ruby/ruby-builder/releases'
 
+const isWin = (os.platform() === 'win32')
+
 export function getAvailableVersions(platform, engine) {
   return rubyBuilderVersions.getVersions(platform)[engine]
 }
 
 export async function install(platform, engine, version) {
   const rubyPrefix = await downloadAndExtract(platform, engine, version)
-  let newPathEntries
-  if (engine === 'rubinius') {
-    newPathEntries = [path.join(rubyPrefix, 'bin'), path.join(rubyPrefix, 'gems', 'bin')]
-  } else {
-    newPathEntries = [path.join(rubyPrefix, 'bin')]
-  }
-  return [rubyPrefix, newPathEntries]
+  return rubyPrefix
 }
 
 async function downloadAndExtract(platform, engine, version) {
-  const rubiesDir = path.join(os.homedir(), '.rubies')
+  const rubiesDir = isWin ?
+    `${(process.env.GITHUB_WORKSPACE || 'C')[0]}:` :
+    path.join(os.homedir(), '.rubies')
+
+  const rubyPrefix = path.join(rubiesDir, `${engine}-${version}`)
+  const newPathEntries = (engine === 'rubinius') ?
+    [path.join(rubyPrefix, 'bin'), path.join(rubyPrefix, 'gems', 'bin')] :
+    [path.join(rubyPrefix, 'bin')]
+
+  common.setupPath(newPathEntries)
```

Comment:
This seems done rather early, even before the `rubyPrefix` exists. I guess we can move it at the end of this function?

Reply:
Ah but for Windows it's useful to add MSYS2 in Path for extraction.

Reply:
> This seems done rather early

I can't recall the timeline of this.  It involves the `afterSetupPathHook`, and the issue of installing packages before `bundle install`.  I don't think I've modified `setup-ruby-pkgs` for that, but I'm not sure.  I'm looking at it shortly.

Reply:
I kept it there, it seems useful notably to get `tar` from MSYS2 on Windows.

Reply:
JFYI, as I recall, that was needed only because the Win2016 image didn't have a native tar, which Win2019 did.  For instance, locally (Windows 10), I have a native Windows `tar`